memcached-operator/pkg/controller/memcached/memcached_controller.go

```diff
@@ -122,15 +130,31 @@ func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Res
 
 	// Ensure the deployment size is the same as the spec
 	size := memcached.Spec.Size
-	if *found.Spec.Replicas != size {
-		found.Spec.Replicas = &size
-		err = r.client.Update(context.TODO(), found)
+	if *deployment.Spec.Replicas != size {
+		deployment.Spec.Replicas = &size
+		err = r.client.Update(context.TODO(), deployment)
 		if err != nil {
-			reqLogger.Error(err, "Failed to update Deployment.", "Deployment.Namespace", found.Namespace, "Deployment.Name", found.Name)
+			reqLogger.Error(err, "Failed to update Deployment.", "Deployment.Namespace", deployment.Namespace, "Deployment.Name", deployment.Name)
 			return reconcile.Result{}, err
 		}
-		// Spec updated - return and requeue
-		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Check if the service already exists, if not create a new one
+	// NOTE: The Service is used to expose the deployment. However, the service is not required at all for the memcached works. The purpose to add it is to bring more clarity over how you can do your own operators-projects.
+	service := &corev1.Service{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: memcached.Name, Namespace: memcached.Namespace}, service)
+	if err != nil && errors.IsNotFound(err) {
+		// Define a new deployment
+		ser := r.serviceForMemcached(memcached)
+		reqLogger.Info("Creating a new Service.", "Service.Namespace", ser.Namespace, "Service.Name", ser.Name)
+		err = r.client.Create(context.TODO(), ser)
+		if err != nil {
+			reqLogger.Error(err, "Failed to create new Service.", "Service.Namespace", ser.Namespace, "Service.Name", ser.Name)
+			return reconcile.Result{}, err
+		}
```

Comment:
Return with requeue here.

Reply:
Why would be required to do the queue here? 

After create the deployment we need to do the requeue in this implementation because we are checking the instance size. However, it also could be solved by fetching the deployment again. However, here has no need to do the requeue. Note that the goal is to show for the user that the requeue is not mandatory at all. See here https://github.com/operator-framework/operator-sdk/issues/1464 for you have a better idea over what motivates me do the PR.

Reply:
Correct me if I'm wrong but if operator state is not as expected and you make an update anywhere in `Reconcile` you should ensure the request is requeued *somewhere*. Your changes do not requeue on any change.

Perhaps having an `var optionalChangeRequeue bool` at the beginning of `Reconcile` that is set to `true` when the Deployment or Service is updated. If true by the end of `Reconcile`, requeue the request.

Reply:
If the state is not as expected will occurs an error then we should always requeue in the errors but not what is the goal of requeue it when the data was created with success?

Reply:
It depends on how important the check is to your operator. There should be one example, ex. the Service, that requeues explicitly with a comment as to why you're requeuing here, and another that does not requeue explicitly, ex the Deployment, with a comment as to why not.

Reply:
Hi @estroz I added a comment which would clarify it. Please, let me know if you are ok with we move forward here.